source/tests/system/nitclk_driver_api_tests.cpp

```diff
@@ -0,0 +1,223 @@
+
+    ::grpc::Status status = GetScopeStub()->Close(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    EXPECT_EQ(kScopeDriverApiSuccess, response.status());
+  }
+
+  ViReal64 get_real64_attribute(const char* channel_name, tclk::NiTClkAttributes attribute_id)
+  {
+    ::grpc::ClientContext context;
+    tclk::GetAttributeViReal64Request request;
+    request.mutable_session()->set_id(GetScopeSessionId());
+    request.set_channel_name(channel_name);
+    request.set_attribute_id(attribute_id);
+    tclk::GetAttributeViReal64Response response;
+
+    ::grpc::Status status = GetTClkStub()->GetAttributeViReal64(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    EXPECT_EQ(kTClkDriverApiSuccess, response.status());
+    return response.value();
+  }
+
+  int get_session_id_from_attribute(const char* channel_name, tclk::NiTClkAttributes attribute_id)
+  {
+    ::grpc::ClientContext context;
+    tclk::GetAttributeViSessionRequest request;
+    request.mutable_session()->set_id(GetScopeSessionId());
+    request.set_channel_name(channel_name);
+    request.set_attribute_id(attribute_id);
+    tclk::GetAttributeViSessionResponse response;
+
+    ::grpc::Status status = GetTClkStub()->GetAttributeViSession(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    EXPECT_EQ(kTClkDriverApiSuccess, response.status());
+    return response.mutable_value()->id();
+  }
+
+  std::string get_string_attribute(const char* channel_name, tclk::NiTClkAttributes attribute_id)
+  {
+    ::grpc::ClientContext context;
+    tclk::GetAttributeViStringRequest request;
+    request.mutable_session()->set_id(GetScopeSessionId());
+    request.set_channel_name(channel_name);
+    request.set_attribute_id(attribute_id);
+    tclk::GetAttributeViStringResponse response;
+
+    ::grpc::Status status = GetTClkStub()->GetAttributeViString(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    EXPECT_EQ(kTClkDriverApiSuccess, response.status());
+    return response.value();
+  }
+
+ private:
+  std::shared_ptr<::grpc::Channel> channel_;
+  std::unique_ptr<::nidevice_grpc::Session> scope_session_;
+  std::unique_ptr<scope::NiScope::Stub> niscope_stub_;
+  std::unique_ptr<tclk::NiTClk::Stub> nitclk_stub_;
+  std::unique_ptr<nidevice_grpc::SessionRepository> session_repository_;
+  std::unique_ptr<scope::NiScopeLibrary> niscope_library_;
+  std::unique_ptr<scope::NiScopeService> niscope_service_;
+  std::unique_ptr<tclk::NiTClkLibrary> nitclk_library_;
+  std::unique_ptr<tclk::NiTClkService> nitclk_service_;
+  std::unique_ptr<::grpc::Server> server_;
+};
+
+TEST_F(NiTClkDriverApiTest, SetAttributeViReal64_GetAttributeViReal64ReturnsSameValue)
+{
+  const char* channel_name = "";
+  const tclk::NiTClkAttributes attribute_to_set = tclk::NiTClkAttributes::NITCLK_ATTRIBUTE_SAMPLE_CLOCK_DELAY;
+  const ViReal64 expected_value = 4.24;
+  ::grpc::ClientContext context;
+  tclk::SetAttributeViReal64Request request;
```

Comment:
Extract setting of attribute and its validation to `set_real64_attribute` method.

Same for other tests below.

Reply:
Done

Reply:
Done